libpoolprotocols/stratum/EthStratumClient.cpp

```diff
@@ -1,1459 +1,1489 @@
+        // Put the actor back to sleep.
+        m_conntimer.async_wait(m_io_strand.wrap(boost::bind(
+            &EthStratumClient::check_connect_timeout, this, boost::asio::placeholders::error)));
+    }
+}
+
+
+void EthStratumClient::connect_handler(const boost::system::error_code& ec)
+{
+    dev::setThreadName("stratum");
+
+    // Set status completion
+    m_conntimer.cancel();
+    m_connecting.store(false, std::memory_order_relaxed);
+
+    // Timeout has run before or we got error
+    if (ec || !m_socket->is_open())
+    {
+        cwarn << ("Error  " + toString(m_endpoint) + " [ " + ( ec ? ec.message() : "Timeout") + " ]");
+
+        // We need to close the socket used in the previous connection attempt
+        // before starting a new one.
+        // In case of error, in fact, boost does not close the socket
+        // If socket is not opened it means we got timed out
+        if (m_socket->is_open())
+            m_socket->close();
+
+        // Discard this endpoint and try the next available.
+        // Eventually is start_connect which will check for an
+        // empty list.
+        m_endpoints.pop();
+        m_canconnect.store(false, std::memory_order_relaxed);
+        m_io_service.post(m_io_strand.wrap(boost::bind(&EthStratumClient::start_connect, this)));
+
+        return;
+    }
+
+    // We got a socket connection established
+    m_canconnect.store(true, std::memory_order_relaxed);
+    if (g_logVerbosity >=6 )
+        cnote << "Socket connected to " << ActiveEndPoint();
+
+    if (m_conn->SecLevel() != SecureLevel::NONE)
+    {
+        boost::system::error_code hec;
+        m_securesocket->lowest_layer().set_option(boost::asio::socket_base::keep_alive(true));
+        m_securesocket->lowest_layer().set_option(tcp::no_delay(true));
+
+        m_securesocket->handshake(boost::asio::ssl::stream_base::client, hec);
+
+        if (hec)
+        {
+            cwarn << "SSL/TLS Handshake failed: " << hec.message();
+            if (hec.value() == 337047686)
+            {  // certificate verification failed
+                cwarn << "This can have multiple reasons:";
+                cwarn << "* Root certs are either not installed or not found";
+                cwarn << "* Pool uses a self-signed certificate";
+                cwarn << "Possible fixes:";
+                cwarn << "* Make sure the file '/etc/ssl/certs/ca-certificates.crt' exists and "
+                         "is accessible";
+                cwarn << "* Export the correct path via 'export "
+                         "SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt' to the correct "
+                         "file";
+                cwarn << "  On most systems you can install the 'ca-certificates' package";
+                cwarn << "  You can also get the latest file here: "
+                         "https://curl.haxx.se/docs/caextract.html";
+                cwarn << "* Disable certificate verification all-together via command-line "

```

Comment:
Does the command-line param still exist? last-time i looked for it, i did not find it anymore.

Reply:
I believe @jean-m-cyr could answer this. He was in charge to migrate command line argument parser.